edb/common/parsing.py

```diff
@@ -97,9 +99,10 @@ def for_lex_token(mcls, token):
 
 
 class Token(parsing.Token, metaclass=TokenMeta):
-    def __init__(self, parser, val, context=None):
+    def __init__(self, parser, val, clean_value, context=None):
```

Comment:
What is `clean_value` and how is it different cfrom `val`?

Reply:
`val` is token text and `clean_value` is the unescaped (for stings) and decoded value (for ints)

Reply:
As long as we don't decode float values, we should be fine.

Reply:
As long as we don't decode float values, we should be fine.

Reply:
Well, currently rust decodes float values too. But we in this change we don't use anything except strings (that's left for to do). So the question is not relevant for this PR, but still what's wrong with decoding floats?

Reply:
Floats are often not representable exactly as a string, which means that the value provided by the user is going to be different from the factual value the float is decoded into. This presents two problems:
1) If you display that float value in an error message it may actually differ from the string the user supplied, which is inconvenient and potentially misleading.
2) If your decoder is different from the one Postgres uses then you'd reconstruct and submit the wrong value. Guaranteeing 100% that you're using the same decoder is probably impractical at this point.

Additionally, there's no reason to decode any numerical literals because we don't actually do any arithmetic on them and potentially this arithmetic may yield different results than Postgres.

We need to clean up the value string to remove the `_` from it, but that's purely a string operation.

Reply:
I think there are reversible algorithms for float to string conversion, but it's not for this PR anyway